### PR TITLE
no exception when no stats folder exists

### LINKED ISSUE #77: [Bug Report] Throws exception if world doesn't have a stats directory
In case a world doesn't have stats saving on and / or simply no stats folder an exception is thrown.

The plugin continues to work, still it would be nicer to handle that case nicer.

I will submit a PR to return an empty hash map in case of no stats folder.

```
[21:18:25 INFO]: [PrometheusExporter] Enabling PrometheusExporter v2.4.0
[21:18:25 INFO]: [PrometheusExporter] Reading player stats from folder  /data/world/stats
[21:18:25 INFO]: [PrometheusExporter] Error - abandoning file reading
[21:18:25 WARN]: java.nio.file.NoSuchFileException: /data/world/stats
[21:18:25 WARN]: 	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
[21:18:25 WARN]: 	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
[21:18:25 WARN]: 	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
[21:18:25 WARN]: 	at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
[21:18:25 WARN]: 	at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149)
[21:18:25 WARN]: 	at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
[21:18:25 WARN]: 	at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
[21:18:25 WARN]: 	at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219)
[21:18:25 WARN]: 	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
[21:18:25 WARN]: 	at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322)
[21:18:25 WARN]: 	at java.base/java.nio.file.FileTreeIterator.<init>(FileTreeIterator.java:71)
[21:18:25 WARN]: 	at java.base/java.nio.file.Files.walk(Files.java:3825)
[21:18:25 WARN]: 	at java.base/java.nio.file.Files.walk(Files.java:3879)
[21:18:25 WARN]: 	at de.sldk.mc.metrics.player.PlayerStatisticLoaderFromFile.readPlayerStatsFiles(PlayerStatisticLoaderFromFile.java:69)
[21:18:25 WARN]: 	at de.sldk.mc.metrics.player.PlayerStatisticLoaderFromFile.<init>(PlayerStatisticLoaderFromFile.java:43)
[21:18:25 WARN]: 	at de.sldk.mc.metrics.PlayerStatistics.<init>(PlayerStatistics.java:40)
[21:18:25 WARN]: 	at de.sldk.mc.config.MetricConfig.getMetric(MetricConfig.java:20)
[21:18:25 WARN]: 	at de.sldk.mc.config.PrometheusExporterConfig.lambda$enableConfiguredMetrics$1(PrometheusExporterConfig.java:62)
[21:18:25 WARN]: 	at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4390)
[21:18:25 WARN]: 	at de.sldk.mc.config.PrometheusExporterConfig.enableConfiguredMetrics(PrometheusExporterConfig.java:61)
[21:18:25 WARN]: 	at de.sldk.mc.PrometheusExporter.onEnable(PrometheusExporter.java:18)
[21:18:25 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[21:18:25 WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380)
[21:18:25 WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[21:18:25 WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501)
[21:18:25 WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415)
[21:18:25 WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:591)
[21:18:25 WARN]: 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:281)
[21:18:25 WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1065)
[21:18:25 WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289)
[21:18:25 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:829)
```

### PR DESCRIPTION
**Pull request by galexrt**
Signed-off-by: Alexander Trost <galexrt@googlemail.com>

Resolves #77

### CODE REVIEW

**Review by weihao — APPROVED** [review]
Solves this error:

```
survival_1       | [07:00:15 INFO]: [PrometheusExporter] Enabling PrometheusExporter v2.4.0
survival_1       | [07:00:15 INFO]: [PrometheusExporter] Reading player stats from folder  /data/world/stats
survival_1       | [07:00:15 INFO]: [PrometheusExporter] Error - abandoning file reading
survival_1       | [07:00:15 WARN]: java.nio.file.NoSuchFileException: /data/world/stats
survival_1       | [07:00:15 WARN]:     at java.base/sun.nio.fs.UnixException.translateToIOException(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/java.nio.file.Files.readAttributes(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/java.nio.file.FileTreeWalker.getAttributes(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/java.nio.file.FileTreeWalker.visit(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/java.nio.file.FileTreeWalker.walk(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/java.nio.file.FileTreeIterator.<init>(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/java.nio.file.Files.walk(Unknown Source)
survival_1       | [07:00:15 WARN]:     at java.base/java.nio.file.Files.walk(Unknown Source)
survival_1       | [07:00:15 WARN]:     at de.sldk.mc.metrics.player.PlayerStatisticLoaderFromFile.readPlayerStatsFiles(PlayerStatisticLoaderFromF
survival_1       | [07:00:15 WARN]:     at de.sldk.mc.metrics.player.PlayerStatisticLoaderFromFile.<init>(PlayerStatisticLoaderFromFile.java:43)
survival_1       | [07:00:15 WARN]:     at de.sldk.mc.metrics.PlayerStatistics.<init>(PlayerStatistics.java:40)
survival_1       | [07:00:15 WARN]:     at de.sldk.mc.config.MetricConfig.getMetric(MetricConfig.java:20)
survival_1       | [07:00:15 WARN]:     at de.sldk.mc.config.PrometheusExporterConfig.lambda$enableConfiguredMetrics$1(PrometheusExporterConfig.ja
survival_1       | [07:00:15 WARN]:     at java.base/java.util.Arrays$ArrayList.forEach(Unknown Source)
survival_1       | [07:00:15 WARN]:     at de.sldk.mc.config.PrometheusExporterConfig.enableConfiguredMetrics(PrometheusExporterConfig.java:61)
survival_1       | [07:00:15 WARN]:     at de.sldk.mc.PrometheusExporter.onEnable(PrometheusExporter.java:18)
survival_1       | [07:00:15 WARN]:     at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
survival_1       | [07:00:15 WARN]:     at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380)
survival_1       | [07:00:15 WARN]:     at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
survival_1       | [07:00:15 WARN]:     at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501)
survival_1       | [07:00:15 WARN]:     at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415)
survival_1       | [07:00:15 WARN]:     at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:591)
survival_1       | [07:00:15 WARN]:     at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:281)
survival_1       | [07:00:15 WARN]:     at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1065)
survival_1       | [07:00:15 WARN]:     at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289)
survival_1       | [07:00:15 WARN]:     at java.base/java.lang.Thread.run(Unknown Source)
survival_1       | [07:00:15 INFO]: [org.eclipse.jetty.server.Server] jetty-11.0.1; built: 2021-02-19T22:03:05.556Z; git: e98092c1791fbda9d157dd53
survival_1       | [07:00:16 INFO]: [org.eclipse.jetty.server.AbstractConnector] Started ServerConnector@2f540ef9{HTTP/1.1, (http/1.1)}{0.0.0.0:92
survival_1       | [07:00:16 INFO]: [org.eclipse.jetty.server.Server] Started Server@7a1c6bab{STARTING}[11.0.1,sto=0] @47444m
```

File: src/main/java/de/sldk/mc/metrics/player/PlayerStatisticLoaderFromFile.java [hunk 1]
```diff
@@ -63,6 +63,10 @@ public PlayerStatisticLoaderFromFile(Plugin plugin) {
             File minecraftDataFolder = plugin.getServer().getWorldContainer().getCanonicalFile();
 
             Path statsFolder = Paths.get(minecraftDataFolder.getAbsolutePath(), "world", "stats");
+            // If the stats folder doesn't exist, don't bother reading it
+            if (!statsFolder.exists()) {
```
Comment (sladkoff) [inline src/main/java/de/sldk/mc/metrics/player/PlayerStatisticLoaderFromFile.java:67]: This doesn't compile on the CI with Java 11. Could you double-check?

File: src/main/java/de/sldk/mc/metrics/player/PlayerStatisticLoaderFromFile.java [hunk 2]
```diff
@@ -63,6 +63,10 @@ public PlayerStatisticLoaderFromFile(Plugin plugin) {
             File minecraftDataFolder = plugin.getServer().getWorldContainer().getCanonicalFile();
 
             Path statsFolder = Paths.get(minecraftDataFolder.getAbsolutePath(), "world", "stats");
+            // If the stats folder doesn't exist, don't bother reading it
```
Comment (sladkoff) [inline src/main/java/de/sldk/mc/metrics/player/PlayerStatisticLoaderFromFile.java:66]: Feel free to drop the comment. The code is pretty self-explanatory :rocket:

**Review by sladkoff — CHANGES_REQUESTED** [review]
@galexrt Thanks for the contribution! Could you check why the CI can't compile this. Cheers.

**Comment by galexrt** [comment]
@sladkoff Fixed.

BTW My IDE is warning me about `Resource leak: 'statFiles' is never closed.` on line 73.

**Comment by sladkoff** [comment]
@galexrt Good catch! Would you mind fixing that up, too :) 
A try-with-resources block should probably work fine there.

**Comment by galexrt** [comment]
@sladkoff Done, pushed in a second commit. :)

**Pull request merged**